// @webriq/gatsby-webriq-form/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

class WebriQForm extends _react.default.Component {
  constructor(props) {
    super(props);
    this.loadWebriQFormScript = this.loadWebriQFormScript.bind(this);
  }

  componentDidMount() {
    if (window && !window.isWebriQFormLoaded) {
      this.loadWebriQFormScript();
    }

    if (window && window.isWebriQFormLoaded) {
      window.webriqFormRefresh();
    }
  }

  loadWebriQFormScript() {
    const script = document.createElement('script');
    script.type = 'text/javascript';
    script.id = 'webriqform';
    script.src = this.props.scriptSrc || 'https://forms.webriq.com/js/initReactForms';
    document.body.appendChild(script);
    const headScript = document.getElementsByTagName('script')[0];
    headScript.parentNode.insertBefore(script, headScript);
  }

  render() {
    const _this$props = this.props,
          id = _this$props.id,
          name = _this$props.name,
          className = _this$props.className;
    const formId = this.props.formId || this.props['data-form-id'];
    let redirectURL = this.props.redirectUrl || this.props['data-thankyou-url'] || '/thank-you';
    return _react.default.createElement("form", {
      name: name,
      id: id,
      className: className,
      method: "POST",
      "data-form-id": formId,
      "data-thankyou-url": redirectURL,
      webriq: "true"
    }, this.props.children);
  }

}

var _default = WebriQForm;
exports.default = _default;
